sdk/client/tx.go

```diff
@@ -101,23 +102,45 @@ func (c *GreenfieldClient) SignTx(ctx context.Context, msgs []sdk.Msg, txOpt *ty
 }
 
 func (c *GreenfieldClient) signTx(ctx context.Context, txConfig client.TxConfig, txBuilder client.TxBuilder, txOpt *types.TxOption) ([]byte, error) {
-	km, err := c.GetKeyManager()
-	if err != nil {
-		return nil, err
+	var km keys.KeyManager
+	var accountNum uint64
+	var account authtypes.AccountI
+	hasOverrideAccount := false
+
+	var err error
+
+	if txOpt == nil || txOpt.OverrideAccount == nil || (*txOpt.OverrideAccount).Km == nil || (*txOpt.OverrideAccount).Num == nil {
+		km, err = c.GetKeyManager()
+		if err != nil {
+			return nil, err
+		}
+		account, err = c.getAccount()
+		if err != nil {
+			return nil, err
+		}
+		accountNum = account.GetAccountNumber()
+	} else {
+		km = *txOpt.OverrideAccount.Km
+		accountNum = *txOpt.OverrideAccount.Num
```

Comment:
`accountNum` seems newly introduced, why don't we only introduce `overide keymanager`, and keep other logic unchanged>